javalib/src/main/scala/java/io/File.scala

```diff
@@ -603,8 +603,7 @@ class File(_path: String) extends Serializable with Comparable[File] {
 
   def deleteOnExit(): Unit = DeleteOnExit.addFile(this.getAbsolutePath())
 
-  @stub
-  def toURL(): java.net.URL = ???
+  // def toURL(): java.net.URL = ???
```

Comment:
> That also means that we would need to implement URI::toURL as well. In my opinion it's fine, as long as we would only the public constructors of URL known from javadocs.

Do you mean something like this: https://github.com/scala-native/scala-native/pull/3348/commits/799464649fc2ae7f39e992597a328e9b2949f83f ?
If I understand correctly, as @ekrich mentioned in https://github.com/scala-native/scala-native/pull/3348#discussion_r1280619728 having `public constructors of URL` in scala-native makes impossible to have a third party URL implementation. And`URI#toURL` will be an URL we defined, not a third-party one.
Am I having a misunderstanding?

Reply:
Ah, you mean, like this? https://github.com/scala-native/scala-native/pull/3348/commits/0f6002a0c6b673695f3f5f7d0ed804c09dfe7527

Reply:
I though it doesn't compile, but actually

- it does compile (to NIR) because the symbol to `java.net.URL` resolves, it is defined in javalib (implemented in Java).
- At linking phase, Reachability check will fail if we call `URI#toURL` because `Not found Top(java.net.URL)`

Am I correct?